slam-ws/src/caffeine/urdf/caffeine.urdf.xacro

```diff
@@ -306,6 +306,49 @@
       <origin xyz="${base_width / 2.0} 0 ${base_height / 2.0 + hokuyo_height / 2.0 + 0.5}" rpy="0 0 ${-135 * pi / 180}"/>
    </joint>
 
+   <!-- ZED Camera -->
+  <link name="zed_camera_link">
+      <visual>
+         <origin xyz="0 0 0" rpy="0 0 0"/>
+            <!-- TODO: should be using the mesh but couldn't get it to work -->
+         <geometry>
+            <box size="${ZED_camera_length} ${ZED_camera_width} ${ZED_camera_height}"/>
+         </geometry>
+       </visual>  
+       <collision>
+           <origin xyz="0 0 0" rpy="0 0 0"/>
+           <geometry>
+               <box size="${ZED_camera_length} ${ZED_camera_width} ${ZED_camera_height}"/>
+           </geometry>
+       </collision>
+       <inertial>
+           <mass value="${ZED_camera_mass}"/> <!-- Obtained from google search -->
+           <xacro:box_inertia mass="${ZED_camera_mass}" length="${ZED_camera_length}" width="${ZED_camera_width}" height="${ZED_camera_height}"/>
+       </inertial>
+   </link>
+   <joint name="zed_camera_joint" type="fixed">
+       <parent link="base_link"/>
+       <child link="zed_camera_link"/>
+       <origin xyz="${stand_length/2} 0 ${stand_height + ZED_camera_height}" rpy="${pi/4} 0 ${pi/2}"/>
+   </joint>
+
+   <!-- ZED camera lenses -->
+   <!-- NOTE: I'm only defining one and not visually showing any-->
+   <!-- I only need one for purposes of using the plugin -->
+   <!-- TODO: check if it's fine to call the left lense camera_link (rtabmap doesn't specify whether it wants the lense frame or the camera frame itself) -->
+   <link name="camera_link"> 
+       <!-- Inertial tag here to ensure that link does not phase through robot-->
+       <inertial>
+            <mass value="0.0001"/>
+            <inertia ixx="0.0001" ixy="0" ixz="0" iyy="0.0001" iyz="0" izz="0.00001"/>
+       </inertial>
+   </link>
+   <joint name="camera_link_joint" type="fixed">
+       <origin xyz="${ZED_camera_length/2.0} 0 0" rpy="0 0 ${3*pi/2}"/>
+       <parent link="zed_camera_link"/>
+       <child link="camera_link"/>
+   </joint>
```

Comment:
Do you not need two lenses? (stereo cam)

Reply:
From what I gathered from @kajananchinniah the tutorial they followed simply used one lens and it works fine because all it is defining is a position and orientation. I tried with two lenses and I couldn't get it to work. I believe he did as well

Reply:
If it becomes an issue later on, then I'll come ask you and @kajananchinniah to look over it. Otherwise it's good for now